### ISSUE TITLE
ld_pruning_api lexical error

### ISSUE THREAD

**Issue by miaokyle** [MRCIEU/TwoSampleMR#74]
Hi,

Got the same issue as other posts.
When calling ld_pruning_api, got the following error message.
Error: lexical error: invalid char in json text.
                                       <!DOCTYPE HTML PUBLIC "-//W3C//
                     (right here) ------^

Some pruning worked but some did not.

**Comment by mightyphil2000** [comment]
Could be a server issue. Have you tried breaking up your query into multiple separate queries? This error is most likely when trying to clump many exposures simultaneously.

________________________________

**Comment by miaokyle** [comment]
Yes, I sent multiple different queries within a short time. Or is there any shortest time limit between two queries?

**Comment by mightyphil2000** [comment]
How many exposures and how many SNPs per exposure are you doing and across how many queries?

________________________________

**Comment by miaokyle** [comment]
There are several thousands queries, and the SNPs number varies, from 1-1000 SNPs, roughly saying. When the less the SNPs in a query, the higher the chance this error would occur. The error happens almost definitely when the number of SNPs is below 10.

**Comment by mightyphil2000** [comment]
Do the SNPs all have rsids?

________________________________

**Comment by miaokyle** [comment]
No, but some queries succeeded even with NAs.

**Comment by mightyphil2000** [comment]
The error is possibly for those exposures where there were no rsids. The function won't work for SNPs without RSids. It's currently RSid dependent.

________________________________

**Comment by miaokyle** [comment]
I do not think that is the solution. After removing all the SNPs without RSid, the same error still happened.

**Comment by miaokyle** [comment]
Tried 'clump_data' and compared the results by using PLINK. If the query set of SNPs have no significant 'clump' results, this error would occur. Looks like TwoSampleMR server did not handle well on the set of SNPs without any significant 'clump' or 'pruning' results.

**Comment by mightyphil2000** [comment]
What happens if you relax the clump p value thresholds?


Could you send the scripts you are using that are returning the errors?

________________________________

**Comment by miaokyle** [comment]
Below is one of data table(df_for_ld_pruning) that failed. You can use this one to test.

> 	SNP	chr_name	chrom_start	pval.exposure	Marker
6973253	rs73880302	22	27382785	2.133362e-10	22:27382785
7015990	rs73882503	22	27382864	2.697752e-10	22:27382864
7067983	rs8137995	22	27376530	3.537212e-10	22:27376530
7146462	rs73882504	22	27385801	5.195226e-10	22:27385801

I used rpy2 to transfer the pandas into R data frame, then fed into the function.

`TwoSampleMR.ld_pruning_api(pandas2ri.py2ri(df_for_ld_pruning))`
